extensions-core/kubernetes-extensions/src/main/java/org/apache/druid/k8s/discovery/K8sDiscoveryModule.java

```diff
@@ -63,7 +64,10 @@ public void configure(Binder binder)
                 try {
                   // Note: we can probably improve things here about figuring out how to find the K8S API server,
                   // HTTP client timeouts etc.
-                  return Config.defaultClient();
+                  final SimpleDateFormat dateFormat = new SimpleDateFormat(
+                      "yyyyMMdd'T'HHmmss.SSS'Z'");
+                  dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
```

Comment:
Why is the timezone specified as UTC? What if users deploy Druid by setting `-Duser.timezone` to some other zones?

Reply:
I parsed out the exact timestamps that are read from the pod metadata.  Where in the codebase is the formatter that sets that?  I can simply use that format to match whatever the user has configured.  I'll look and see if I can find it, but if you happen to know where to look I'd appreciate it.

Reply:
So the formatter is used to parsed the date string that are returned from K8S server side? And the string is always in UTC format? If so, the change makes senses.

Reply:
It's the kind of thing that should be always UTC, since it's a server thing, not a user-facing or data-related timestamp.

Reply:
I need to confirm that this _always_ originates in K8s, and is always consistent.  It is likely better to allow this format string to be configured by the user, but to default the most common format.